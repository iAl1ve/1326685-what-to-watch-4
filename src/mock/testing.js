export const ListMovies = [
  {
    id: 1,
    title: `The Grand Budapest Hotel`,
    genre: `Comedy`,
    year: 1996,
    src: `img/the-grand-budapest-hotel-poster.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `8.9`,
    ratingLevel: `Very good`,
    ratingCount: 256,
    movieDescription: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    movieDirector: `Director: Wes Andreson`,
    movieStarring: `Starring: Bill Murray, Edward Norton, Jude Law, Willem Dafoe and other`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: `1h 44m`,
  }, {
    id: 2,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `drama`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `9.1`,
    ratingLevel: `Very good +`,
    ratingCount: 146,
    movieDescription: `Fantastic Beasts: The Crimes of Grindelwald movie description text. Fantastic Beasts: The Crimes of Grindelwald movie description text. Fantastic Beasts: The Crimes of Grindelwald movie description text. `,
    movieDirector: `Wes Andreson`,
    movieStarring: `Bill Murray, Willem Dafoe and other`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: `1h 44m`,
  }, {
    id: 3,
    title: `Bohemian Rhapsody`,
    src: `img/bohemian-rhapsody.jpg`,
    genre: `comedy`,
    year: 2015,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `4.1`,
    ratingLevel: `Very poor `,
    ratingCount: 195,
    movieDescription: `Bohemian Rhapsody movie description text. Bohemian Rhapsody movie description text. Bohemian Rhapsody movie description text. `,
    movieDirector: `Bill Murray`,
    movieStarring: `Edward Norton, Willem Dafoe and other`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: `1h 53m`,
  }, {
    id: 4,
    title: `Macbeth`,
    src: `img/macbeth.jpg`,
    genre: `drama`,
    year: 2014,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `9.3`,
    ratingLevel: `Very poor `,
    ratingCount: 119,
    movieDescription: `Macbeth movie description text. Macbeth movie description text. Macbeth movie description text. Macbeth movie description text. `,
    movieDirector: `Macbeth`,
    movieStarring: `Macbeth, Willem Dafoe and other`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: `1h 56m`,
  }, {
    id: 5,
    title: `Aviator`,
    src: `img/aviator.jpg`,
    genre: `documentary`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `9.2`,
    ratingLevel: `Good`,
    ratingCount: 249,
    movieDescription: `Aviator movie description text. Aviator movie description text. Aviator movie description text. Aviator movie description text.`,
    movieDirector: `Aviator`,
    movieStarring: `Aviator, Willem Dafoe and other`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: `1h 31m`,
  }, {
    id: 6,
    title: `Moonrise Kingdom`,
    src: `img/moonrise-kingdom.jpg`,
    genre: `romance`,
    year: 2011,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `9.1`,
    ratingLevel: `Good`,
    ratingCount: 232,
    movieDescription: `Moonrise Kingdom movie description text. Moonrise Kingdom movie description text. Moonrise Kingdom movie description text. Moonrise Kingdom movie description text.`,
    movieDirector: `Moonrise Kingdom`,
    movieStarring: `Moonrise Kingdom, Willem Dafoe and other`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: `1h 39m`,
  }, {
    id: 7,
    title: `Midnight Special`,
    src: `img/midnight-special.jpg`,
    genre: `crime`,
    year: 2016,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `8.4`,
    ratingLevel: `Good +`,
    ratingCount: 212,
    movieDescription: `Midnight Special movie description text. Midnight Special movie description text. Midnight Special movie description text. Midnight Special movie description text.`,
    movieDirector: `Midnight Special`,
    movieStarring: `Midnight Special, Willem Dafoe and other`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: `2h 34m`,
  }, {
    id: 8,
    title: `War of the Worlds`,
    src: `img/war-of-the-worlds.jpg`,
    genre: `drama`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `8.9`,
    ratingLevel: `Very good`,
    ratingCount: 201,
    movieDescription: `War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. `,
    movieDirector: `War of the Worlds`,
    movieStarring: `War of the Worlds, Willem Dafoe and other`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: `1h 39m`,
  }, {
    id: 9,
    title: `Shutter Island`,
    src: `img/shutter-island.jpg`,
    genre: `drama`,
    year: 2017,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `7.9`,
    ratingLevel: `Good`,
    ratingCount: 175,
    movieDescription: `Shutter Island movie description text. Shutter Island movie description text. Shutter Island movie description text. Shutter Island movie description text. `,
    movieDirector: `Shutter Island`,
    movieStarring: `Shutter Island, Willem Dafoe and other`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: `1h 56m`,
  }
];

export const ListReviews = [
  {
    id: 1,
    idFilm: 1,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    date: `2016-12-24`,
    rating: `7,6`,
  }, {
    id: 2,
    idFilm: 1,
    text: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    author: `Matthew Lickona`,
    date: `2016-12-20`,
    rating: `8,3`,
  }, {
    id: 3,
    idFilm: 1,
    text: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    author: `Bill Goodykoontz`,
    date: `2015-11-18`,
    rating: `7,3`,
  },
  {
    id: 4,
    idFilm: 2,
    text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    author: `Paula Fleri-Soler`,
    date: `2016-12-20`,
    rating: `7,8`,
  },
  {
    id: 5,
    idFilm: 2,
    text: `When one of Gustave's lovers dies mysteriously. It is certainly a magical and childlike way of storytelling.`,
    author: `Amanda Soler`,
    date: `2016-12-20`,
    rating: `9,6`,
  },
  {
    id: 6,
    idFilm: 3,
    text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    author: `Amanda Greever`,
    date: `2015-11-18`,
    rating: `8,9`,
  },
  {
    id: 7,
    idFilm: 4,
    text: `Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Bill Muir`,
    date: `2016-08-14`,
    rating: `8,4`,
  },
  {
    id: 8,
    idFilm: 5,
    text: `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs.`,
    author: `Wes  Dafoe`,
    date: `2015-11-21`,
    rating: `5,6`,
  },
  {
    id: 9,
    idFilm: 2,
    text: `Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    author: `Kate Norton`,
    date: `2015-03-08`,
    rating: `2`,
  },
  {
    id: 10,
    idFilm: 1,
    text: `Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    author: `Andreson Fleri`,
    date: `2016-05-09`,
    rating: `10`,
  },
];
