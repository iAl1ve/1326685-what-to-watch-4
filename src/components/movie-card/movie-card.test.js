import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card.jsx";
import {ListMovies} from "../../mock/testing.js";

const onTitleButtonClick = () => {};
const onMovieCardHover = () => {};
const onMouseCardLeave = () => {};

describe(`Test Render MovieCard`, () => {
  it(`Render MovieCard`, () => {
    const tree = renderer
      .create(<MovieCard
        movie = {ListMovies[0]}
        isPlaying = {false}
        onTitleButtonClick = {onTitleButtonClick}
        onMovieCardHover = {onMovieCardHover}
        onMouseCardLeave = {onMouseCardLeave}
      />, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
