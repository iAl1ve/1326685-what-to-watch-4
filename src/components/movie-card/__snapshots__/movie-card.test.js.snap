// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Render MovieCard Render MovieCard 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <div
    className="small-movie-card__image"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="The Grand Budapest Hotel"
      height="175"
      src="img/the-grand-budapest-hotel-poster.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      The Grand Budapest Hotel
    </a>
  </h3>
</article>
`;
